UI: Adding TexturedButton2D and TextButton2D

ganimtron-10 opened the pull request:
Adding Button2D abstract with TexturedButton2D and TextButton2D

<img width="1440" height="1510" alt="image" src="https://github.com/user-attachments/assets/972fc662-4d36-403c-a58b-4b3ae07962ff" />

maharshi-gor commented on fury/ui/core.py:
```diff
@@ -1381,222 +1377,222 @@ def _get_size(self):
+            actor.add_event_handler(self._handle_enter, "pointer_enter")
+            actor.add_event_handler(self._handle_leave, "pointer_leave")
+            actor.add_event_handler(self._handle_down, "pointer_down")
+            actor.add_event_handler(self._handle_up, "pointer_up")
```
This is already being done in UI I think we are repeating it and it can be resused better.

maharshi-gor commented on docs/examples/viz_button.py:
```diff
@@ -0,0 +1,64 @@
+"""
+
+##############################################################################
+# First, a bunch of imports
```
Write this nicely.

ganimtron-10 replied:
This follows the same format as every other demo

maharshi-gor commented on fury/ui/tests/test_elements.py:
```diff
@@ -1,1443 +1,1553 @@
-from os.path import join as pjoin
-from tempfile import TemporaryDirectory as InTemporaryDirectory
-
+# import os
```
Feel free to delete the stuff which not getting used. We will fetch it from old commit if we like to no commented code.

maharshi-gor reviewed (changes requested):
Hello @ganimtron-10  thanks for the work. The PR works nicely.

Let's try to look into events binding. And some easy comments to fix.

maharshi-gor commented:
Hi @ganimtron-10 please rebase this PR.

ganimtron-10 committed:
feat: adding Button2D

feat: adding Button2D

temp: changes

fix: updating buton to be reactive

fix: update button icons on states

fix: moving actual buttons to elements

chore: adding button tests

ganimtron-10 commented on fury/ui/elements.py:
```diff
@@ -20,444 +23,233 @@
+            font_size=self.font_size,
+            size=self._dims,
+        )
+        self.handle_events(self.child.actor)
+        self.handle_events(self.child.background.actor)
```
The `handle_events` should be refactored/rearchitected to enhance DX, maybe by allowing UI or actor or actors so that we dont need to manually specify all actors that to pickup specific events.

ganimtron-10 commented:
Hi @maharshi-gor,
PTAL, updated and rebased.

maharshi-gor commented:
Thanks for the changes @ganimtron-10
